ui/src/shared/components/tables/TableCell.tsx

```diff
@@ -33,14 +33,15 @@ interface Props extends CellRendererProps {
 
 class TableCell extends PureComponent<Props> {
   public render() {
-    const {rowIndex, columnIndex, onHover} = this.props
+    const {data, rowIndex, columnIndex, onHover} = this.props
     return (
       <div
         style={this.style}
         className={this.class}
         onClick={this.handleClick}
         data-column-index={columnIndex}
         data-row-index={rowIndex}
+        {...(rowIndex === 0 ? {'data-testID': `${data}-table-header`} : {})} // conditionally adds test-ID for tableCell header
```

Comment:
dang this is ugly

Reply:
Ya but it’s less that conditionally rendering a component with 1 added prop / adding a testID unnecessarily and sullying the DOM